File: src/test/java/com/jetbrains/space/page/widget/BottomNavBar.java
package com.jetbrains.space.page.widget;

import io.appium.java_client.pagefactory.AndroidFindBy;
import io.appium.java_client.pagefactory.Widget;
import io.qameta.allure.Step;
import org.openqa.selenium.WebElement;

@AndroidFindBy(id = "bottomNavigationView")
public class BottomNavBar extends Widget {

    @AndroidFindBy(id = "chats")
    private WebElement chatsButton;

    @AndroidFindBy(id = "gotoany")
    private WebElement searchButton;

    @AndroidFindBy(id = "todo")
    private WebElement todoButton;

    @AndroidFindBy(id = "dashboard")
    private WebElement dashboardButton;

    protected BottomNavBar(WebElement element) {
        super(element);
    }

    @Step("Navigate to chats")
    public void navigateToChats() {
        chatsButton.click();
    }

    @Step("Navigate to searsh")
    public void navigateToSearch() {
        searchButton.click();
    }

    @Step("Navigate to todo")
    public void navigateToTodo() {
        todoButton.click();
    }

    @Step("Navigate to dashboard")
    public void navigateToDashboard() {
        dashboardButton.click();
    }
}

File: src/test/kotlin/com/jetbrains/space/page/BottomNavBar.kt
package com.jetbrains.space.page

import io.appium.java_client.pagefactory.AndroidFindBy
import io.appium.java_client.pagefactory.Widget
import org.openqa.selenium.WebElement

@AndroidFindBy(id = "bottomNavigationView")
class BottomNavBar(element: WebElement): Widget(element) {

    @AndroidFindBy(id = "chats")
    lateinit var chatsButton: WebElement;

    @AndroidFindBy(id = "gotoany")
    lateinit var searchButton: WebElement;

    @AndroidFindBy(id = "todo")
    lateinit var todoButton: WebElement;

    @AndroidFindBy(id = "dashboard")
    lateinit var dashboardButton: WebElement;
}